cookbooks/aws-parallelcluster-platform/resources/enroot/partial/_enroot_common.rb

```diff
@@ -18,50 +18,36 @@
 action :setup do
   return if on_docker?
   action_install_package
-end
-
-action :configure do
-  return if on_docker?
-  return unless enroot_installed
 
-  cookbook_file "/tmp/enroot.template.conf" do
-    source 'enroot/enroot.template.conf'
-    cookbook 'aws-parallelcluster-platform'
+  template "/etc/enroot/enroot.conf" do
+    source 'enroot/enroot.conf.erb'
     owner 'root'
     group 'root'
-    mode '0755'
-    action :create_if_missing
+    mode '0644'
   end
 
-  bash "Configure enroot" do
-    user 'root'
-    code <<-ENROOT_CONFIGURE
-      set -e
-      ENROOT_CONFIG_RELEASE=pyxis
-      SHARED_DIR=#{node['cluster']['shared_dir']}
-      NONROOT_USER=#{node['cluster']['cluster_user']}
-      mkdir -p ${SHARED_DIR}/enroot
-      chown ${NONROOT_USER} ${SHARED_DIR}/enroot
-      ENROOT_CACHE_PATH=${SHARED_DIR}/enroot envsubst < /tmp/enroot.template.conf > /tmp/enroot.conf
-      mv /tmp/enroot.conf /etc/enroot/enroot.conf
-      chmod 0644 /etc/enroot/enroot.conf
-
-      mkdir -p /tmp/enroot
-      chmod 1777 /tmp/enroot
-      mkdir -p /tmp/enroot/data
-      chmod 1777 /tmp/enroot/data
-
-      chmod 1777 ${SHARED_DIR}/enroot
+  directory node['cluster']['enroot']['persistent_dir'] do
```

Comment:
do we want to keep the mode as 1777? or 755?

Reply:
I'm aligned with your security concern. However, we should use `1777` by design because enroot writes in there as whatever user submitting a containerized job.

Example of files written by enroot in `/var/enroot` after a containerized job is submitted as ec2-user:

```
[root@q1-st-cr1-1 ~]# tree /var/enroot
tree /var/enroot
/var/enroot
└── cache
    └── group-1000
        ├── 6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac
        └── 97271d29cb7956f0908cfb1449610a2cd9cb46b004ac8af25f0255663eb364ba

2 directories, 2 files
[root@q1-st-cr1-1 ~]# ls -la /var/enroot
total 0
drwxrwxrwt  3 root     root      19 Oct 15 15:26 .
drwxr-xr-x 22 root     root     307 Oct 15 11:22 ..
drwx------  3 ec2-user ec2-user  24 Oct 15 15:26 cache
[root@q1-st-cr1-1 ~]# ls -la /var/enroot/cache
total 0
drwx------ 3 ec2-user ec2-user  24 Oct 15 15:26 .
drwxrwxrwt 3 root     root      19 Oct 15 15:26 ..
drwx------ 3 ec2-user ec2-user 170 Oct 15 15:26 group-1000
[root@q1-st-cr1-1 ~]# ls -la /var/enroot/cache/group-1000/
total 29848
drwx------ 3 ec2-user ec2-user      170 Oct 15 15:26 .
drwx------ 3 ec2-user ec2-user       24 Oct 15 15:26 ..
-rw-r----- 1 ec2-user ec2-user 30559455 Oct 15 15:26 6414378b647780fee8fd903ddb9541d134a1947ce092d08bdeb23a54cb3684ac
-rw-r----- 1 ec2-user ec2-user      806 Oct 15 15:26 97271d29cb7956f0908cfb1449610a2cd9cb46b004ac8af25f0255663eb364ba
drwx------ 2 ec2-user ec2-user        6 Oct 15 15:31 .tokens.1000
```

Reply:
Agreed